src/stdlib/builtin.ts

```diff
@@ -471,7 +489,7 @@ function fraction(
     node.numerator *= factor;
     node.denominator = denominator;
   }
-  return new Interval(value, 'linear', node, interval);
+  return new Interval(value, 'linear', interval.steps, node, interval);
```

Comment:
Including steps is a bad idea because it breaks the correspondence between node and value.